.github/workflows/main.yml

```diff
@@ -2,24 +2,50 @@ name: CI
 
 on:
   push:
-    branches: "*"
   pull_request:
-    branches: "*"
 
 jobs:
+  check_duplicate_runs:
```

Comment:
There is an annoying bug of Github Actions - when someone creates a new branch and pushes some commits here, and then creates a pull request, there are 2 copies of workflow running - one for a push event and another for created PR.

This step prevents the second copy of workflow to be run - if tests are already passed for push event then they will be not run twice while PR is created. Also they will not be run if source code hasn't been changed.